code/game/machinery/kitchen/cooking_machines/_cooker.dm

```diff
@@ -1,236 +1,145 @@
-			if(!cooking || !result || result.loc != src)
-				failed = 1
-			else if(prob(burn_chance))
-				// You dun goofed.
-				qdel(cooking_obj)
-				cooking_obj = new /obj/item/weapon/reagent_containers/food/snacks/badrecipe(src)
-				// Produce nasty smoke.
-				visible_message("<span class='danger'>\The [src] vomits a gout of rancid smoke!</span>")
-				var/datum/effect/effect/system/smoke_spread/bad/smoke = getFromPool(/datum/effect/effect/system/smoke_spread/bad)
-				smoke.attach(src)
-				smoke.set_up(10, 0, usr.loc)
-				smoke.start()
-				failed = 1
-
-			if(failed)
-				cooking = 0
-				icon_state = off_icon
-				break
-
-/obj/machinery/cooker/proc/check_cooking_obj()
-	if(!cooking_obj || cooking_obj.loc != src)
-		cooking_obj = null
-		icon_state = off_icon
-		cooking = 0
-		return 0
-	return 1
-
-/obj/machinery/cooker/attack_hand(var/mob/user)
-
-	if(cooking_obj)
-		user << "<span class='notice'>You grab \the [cooking_obj] from \the [src].</span>"
-		user.put_in_hands(cooking_obj)
-		cooking = 0
-		cooking_obj = null
-		icon_state = off_icon
-		return
-
-	if(output_options.len)
-
-		if(cooking)
-			user << "<span class='warning'>\The [src] is in use!</span>"
-			return
-
-		var/choice = input("What specific food do you wish to make with \the [src]?") as null|anything in output_options+"Default"
-		if(!choice)
-			return
-		if(choice == "Default")
-			selected_option = null
-			user << "<span class='notice'>You decide not to make anything specific with \the [src].</span>"
-		else
-			selected_option = choice
-			user << "<span class='notice'>You prepare \the [src] to make \a [selected_option].</span>"
+		light = image(icon, "light_off")
+	light.pixel_x = light_x
+	light.pixel_y = light_y
+	overlays += light
+
 
+
+
+/obj/machinery/appliance/cooker/process()
+	if (!stat)
+		heat_up()
+
+	else
+		var/turf/T = get_turf(src)
+		if (temperature > T.temperature)
+			equalize_temperature()
 	..()
 
-/obj/machinery/cooker/proc/cook_mob(var/mob/living/victim, var/mob/user)
-	return
+/obj/machinery/appliance/cooker/power_change()
+	.=..()
+	update_icon()
+
+/obj/machinery/appliance/cooker/proc/update_cooking_power()
+	var/temp_scale = 0
+	if(temperature > min_temp)
+
+		temp_scale = (temperature - min_temp) / (optimal_temp - min_temp)
+		//If we're between min and optimal this will yield a value in the range 0-1
 
-/obj/machinery/cooker/proc/change_product_strings(var/obj/item/weapon/reagent_containers/food/snacks/product)
-	if(product.type == /obj/item/weapon/reagent_containers/food/snacks/variable) // Base type, generic.
-		product.name = "[cook_type] [cooking_obj.name]"
-		product.desc = "[cooking_obj.desc] It has been [cook_type]."
+		if (temp_scale > 1)
+			//We're above optimal, efficiency goes down as we pass too much over it
+			if (temp_scale >= 2)
+				temp_scale = 0
+			else
+				temp_scale = 1 - (temp_scale - 1)
+
+
+	cooking_power = optimal_power * temp_scale
+
+	//cooking_power *= 1000 //This is for debugging, must remove
```

Comment:
power is still used as long as the temperature is below target. Cooking power is unrelated to power use. It's a measure of how fast it cooks things